src/componenets/taskItem.jsx

```diff
@@ -1,27 +1,46 @@
-import React from 'react'
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { fetch } from "../utils";
 
-const TaskItem = ({ description }) => {
-    return (
-        <div className="task">
-            <p>{description}</p>
-                <div>
-                  <button>
-                    <svg
-                      className="delete"
-                      stroke="currentColor"
-                      fill="#eb6d7d"
-                      strokeWidth="0"
-                      viewBox="0 0 448 512"
-                      height="1em"
-                      width="1em"
-                      xmlns="http://www.w3.org/2000/svg"
-                    >
-                      <path d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"></path>
-                    </svg>
-                  </button>
-                </div>
-        </div>
-    )
-}
+const accessToken = localStorage.getItem("accessToken");
 
-export default TaskItem
+const TaskItem = ({ description, id }) => {
+    const history = useHistory()
+  const deleteTask = () => {
```

Comment:
If you have 500 tasks this deleteTask function would be created in computer memory 500 times because the TaskItem component is created 500 times, this takes up computer memory which is not good. Its better to define this function in the TodoApp component and then pass it down into the TaskItem component as a prop, that way it is created only once in memory and its reference is passed down to the 500 TaskItem instances.